### PR TITLE
122 WIP upgrade some minor packages.

### PR DESCRIPTION
**Pull request by mikelax**
sadface trying to upgrade apollo and graphql (tag). left comment in #122 with error, will have to try other lib combinations. 
Addresses #122

### CODE REVIEW

**Comment by mikelax** [comment]
@nazar  I got all the apollo and react libs upgraded. 
So there is an [open ticket](https://github.com/apollographql/graphql-tag/issues/155) where graphql-tag does not work with graphql > 0.13.0

Also, I was able to just use the apollo-boost lib instead of bringing in all the libs separately. There are some peer warnings, its like a spiders web trying to get all the various lib versions to work together.

I think this is ready to go. Then I'll do a separate PR to upgrade the cra libs, and then one for the remaining dev deps like eslint, etc.

**Comment by mikelax** [comment]
@nazar  ok I got the eslint upgrades done, and fixed a few errors in api. 
If you want a facepalm, here is the reason the jsx-a11y lib isn't updated in the client folder. 
TLDR, cra has issues using esllint, even after ejecting.... 😕 
https://github.com/facebook/create-react-app/issues/3418
https://github.com/odota/web/issues/1020#issuecomment-381165135
https://github.com/evcohen/eslint-plugin-jsx-a11y/issues/397#issuecomment-362839806

This is ready to be merged though. 👍

**Review by nazar — APPROVED** [review]
:shipit:  👍  💯
